web/src/api.tsx

```diff
@@ -84,8 +87,8 @@ export async function getExecutionStatus(id: string): Promise<object> {
 export async function startExecution(
   id: string,
   formData: FormData,
-): Promise<object> {
-  return tryFetchJson(`execution/${id}/start`, {
+): Promise<Response> {
+  return tryFetchApi(`execution/${id}/start`, {
```

Comment:
For the mobile api we decided not to put any request parameter to the url to keep the url clean.
I would suggest you use the same standard and use query parameters instead of path variables, especially for POST request.

Reply:
Since we already decided on that, we should probably stick to that. However, I am not a fan of doing stuff like that via query parameters. 

Usually, in REST apis, the path is used to identify a resource and the query is used to, for example, provide filter criteria (e.g., in a search).

Reply:
Generally, I think we should stick more to the semantics of REST. 

I dont really see the point of having an extra "start" endpoint - we should use a simple `POST` request against `/execution/{id}`. As a counterpart, `DELETE /execution/{id}` should stop the execution. 

Creating an execution should be `POST /execution`, while `GET /execution` retrieves all executions and `GET /execution/{id}` retrieves the execution with the provided id.

Reply:
"I don't remember agreeing on anything and would therefore advocate for using path parameters." 
This is a very critical call to make. Maybe you missed the discussion due to your illness or less interests in the discussion because we were discussing the interface to the mobile api and was therefore less relevant. I was just pointing out the facts we have already implemented so far.

I see the Clean URL reference and will revoke my suggestion. However i do not agree to use POST execution as start endpoint, as well as DELETE to end execution. This leads to further confusion in the semantics and the current willingnes to document the pushed code supports that statement.

Reply:
@YIllmann I think, we should discuss the semantics of our API at a later point and maybe change the endpoints in a future PR. For now, this is fine for me.